crates/ra_assists/src/handlers/add_new.rs

```diff
@@ -42,50 +42,56 @@ pub(crate) fn add_new(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
     let impl_def = find_struct_impl(&ctx, &strukt)?;
 
     let target = strukt.syntax().text_range();
-    acc.add(AssistId("add_new"), "Add default constructor", target, |builder| {
-        let mut buf = String::with_capacity(512);
-
-        if impl_def.is_some() {
-            buf.push('\n');
-        }
-
-        let vis = strukt.visibility().map_or(String::new(), |v| format!("{} ", v));
-
-        let params = field_list
-            .fields()
-            .filter_map(|f| {
-                Some(format!("{}: {}", f.name()?.syntax(), f.ascribed_type()?.syntax()))
-            })
-            .sep_by(", ");
-        let fields = field_list.fields().filter_map(|f| f.name()).sep_by(", ");
-
-        format_to!(buf, "    {}fn new({}) -> Self {{ Self {{ {} }} }}", vis, params, fields);
-
-        let start_offset = impl_def
-            .and_then(|impl_def| {
+    acc.add(
+        AssistId("add_new"),
+        AssistKind::Refactor,
```

Comment:
Let's add `AssistKind::Generate` for things which generate new code (add constructor, add impl)

Reply:
I'm now thinking that there's no reason to do this and we should just leave this kind as empty since there's no UI need to distinguish this case.

Reply:
Heh, my reason for suggesting this was that IntelliJ actually has a dedicated UI for generating stuff: https://www.jetbrains.com/help/idea/generating-code.html

Reply:
Definitely. I think if we get a menu to do that then we should re-categorize.